docs/operator-dependencies-and-requirements.md

```diff
@@ -0,0 +1,35 @@
+# Operator Dependency and Requirement Resolution
+
+## Overview on Dependency Principles
+
+OLM manages the dependency resolution and upgrade lifecycle of running operators.
+
+An operator's dependencies are defined through the (Group, Version, Kind) of CustomResourceDefinitions (CRDs). An operator cannot define a dependency on a specific version (e.g. `etcd-operator v0.9.0`) or instance (e.g. `etcd v3.2.1`) of an operator. This encourages operator authors to depend on the interface and not the implementation; thereby allowing operators to update on individual release cycles.
+
+OLM manages operator dependency resolution very similarily to OS package managers, such as apt/dkpg and yum/rpm. However, given that operators are always running, OLM attempts to ensure that it never deploys a combination of operators that do not work with each other.
+
+This means that OLM will never:
+
+* install a set of operators that require APIs that can't be provided
+* update an operator in a way that breaks another that depends upon it
+
+### Dependency Resolution Alogorithm
+
+Dependency resolution begins when a `subscription` is reconciled by OLM.
+
+When resolving a subscription, OLM will look at all subscriptions in the namespace and identify the latest offering of each operator. OLM will consider if any of these operators require an API that is missing from the cluster. For each of these missing APIs, OLM will select an operator that requires the missing API and prioritize its `CatalogSource` to search for the missing API. If the prioritized `CatalogSource` does not contain the API, OLM will search through the remaining `CatalogSources` in the namespace for the API. If an operator is found that provides the API, OLM will create a subscription for the operator using that `CatalogSource`.
```

Comment:
> identify the latest offering of each operator

Might want to be careful about "latest" here - OLM will find the "next" update for the operator, which is controlled by the update graph (either an operator `replaces` or `skips` the current one, or there's an operator in the head of the channel with a `skipRange`)

Reply:
Addressed in the latest version of the PR.